specification/powerbidedicated/Powerbidedicated.Management/models.tsp

```diff
@@ -0,0 +1,692 @@
+  Key: "Key",
+}
+
+/**
+ * The name of the Azure pricing tier to which the SKU applies.
+ */
+union VCoreSkuTier {
+  string,
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  AutoScale: "AutoScale",
+}
+
+/**
+ * The current deployment state of an auto scale v-core resource. The provisioningState is to indicate states for resource provisioning.
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+union VCoreProvisioningState {
+  string,
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Succeeded: "Succeeded",
+}
+
+/**
+ * Represents the SKU name and Azure pricing tier for PowerBI Dedicated capacity resource.
+ */
+model CapacitySku {
+  /**
+   * Name of the SKU level.
+   */
+  name: string;
+
+  /**
+   * The name of the Azure pricing tier to which the SKU applies.
+   */
+  tier?: CapacitySkuTier;
+
+  /**
+   * The capacity of the SKU.
+   */
+  capacity?: int32;
+}
+
+/**
+ * Properties of Dedicated Capacity resource.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model DedicatedCapacityProperties extends DedicatedCapacityMutableProperties {
+  /**
+   * The current state of PowerBI Dedicated resource. The state is to indicate more states outside of resource provisioning.
+   */
+  @visibility(Lifecycle.Read)
+  state?: State;
+
+  /**
+   * The current deployment state of PowerBI Dedicated resource. The provisioningState is to indicate states for resource provisioning.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: CapacityProvisioningState;
+}
+
+/**
+ * An object that represents a set of mutable Dedicated capacity resource properties.
+ */
+model DedicatedCapacityMutableProperties {
+  /**
+   * A collection of Dedicated capacity administrators
+   */
+  administration?: DedicatedCapacityAdministrators;
+
+  /**
+   * Specifies the generation of the Power BI Embedded capacity. If no value is specified, the default value 'Gen2' is used. [Learn More](https://docs.microsoft.com/power-bi/developer/embedded/power-bi-embedded-generation-2)
+   */
+  mode?: Mode;
+
+  /**
+   * Tenant ID for the capacity. Used for creating Pro Plus capacity.
+   */
+  @visibility(Lifecycle.Read)
+  tenantId?: string;
+
+  /**
+   * Capacity name
+   */
+  @visibility(Lifecycle.Read)
+  friendlyName?: string;
+}
+
+/**
+ * An array of administrator user identities
+ */
+model DedicatedCapacityAdministrators {
+  /**
+   * An array of administrator user identities.
+   */
+  members?: string[];
+}
+
+/**
+ * Represents an instance of an PowerBI Dedicated resource.
+ */
+model Resource {
```

Comment:
>Resource [](http://example.com/codeflow?start=6&length=8)

why are we defining 'Resource' type here, instead of using standard Azure resource model?

Reply:
Question I'd like to raise why aren't we autodetecting to reuse common types here, or at least having nice automatic comments that justify not doing it on the basis of a breaking change (and what change)